### ISSUE TITLE
Implement new AAD Rego policy MS.AAD.3.3v1 MS authenticator context information

### ISSUE THREAD

**Issue by tkol2022** [cisagov/ScubaGear#410]
# 💡 Summary #

Implement a new Rego policy check for MS.AAD.3.3v1 If Phishing-resistant MFA has not been enforced yet and Microsoft Authenticator is enabled, it SHALL be configured to show login context information.

- If policy MS.AAD.3.1v1 Passes, then this policy MS.AAD.3.3v1 will automatically Pass. See the way the OneDrive expiration date for Anyone links is dependent on a different setting for an example.
- You must also check if MS Authenticator is enabled on the Azure Active Directory > Security > Authentication methods page. If Authenticator is not enabled, then this policy will automatically Pass since there is no reason to check the configuration settings.
- If Authenticator is enabled, then you need to check to ensure the [Authenticator settings match the values detailed in the implementation instructions](https://github.com/cisagov/ScubaGear/blob/emerald/baselines/aad.md#msaad33v1-instructions).

## Acceptance criteria ##

How do we know when this work is done?

- [ ] Test the code for all possible value combinations of MS.AAD.3.1v1 and MS.AAD.3.3v1 to ensure that the report produces an accurate Pass or Fail. You must also test scenarios where the MS Authenticator authentication method is Enabled and Disabled to ensure the results product expected evaluations.
- [ ] Test in all tenants

**Comment by tkol2022** [comment]
@schrolla Did some hands-on research on this one. It can be assigned to someone now. Whoever starts working on this should check in with me to get some guidance.

Turns out an existing Cmdlet in the AAD provider can be used to get the data needed for this issue: **Get-MgBetaPolicyAuthenticationMethodPolicy** but the code might need to be adjusted to get the cmdlet to expose the data needed for this issue and Crutch recently modified the respective provider line of code in PR #902 so whoever works on this will need to work from Crutch's version to eliminate conflicts.

Here is an example of how to check whether the MS authenticator is enabled.
`(Get-MgBetaPolicyAuthenticationMethodPolicy).authenticationMethodConfigurations | fl`

![image](https://github.com/cisagov/ScubaGear/assets/107076927/2008a367-b0b8-4906-acd0-e1473ef68229)

Here is an example of how to interrogate the configuration settings of MS authenticator to see if **show application name** and **show geographic location** configurations match the baseline:

![image](https://github.com/cisagov/ScubaGear/assets/107076927/694affcc-ef35-41b3-8674-3b5da6aa0283)

`((Get-MgBetaPolicyAuthenticationMethodPolicy).authenticationMethodConfigurations | Where-Object { $_.Id -eq 'MicrosoftAuthenticator' }).AdditionalProperties["featureSettings"] | fl`

![image](https://github.com/cisagov/ScubaGear/assets/107076927/bdc06ed8-a697-44b2-b659-f5b56452b5ae)

For the feature settings **displayAppInformationRequiredState** and **displayLocationInformationRequiredState** you have to make sure they are enabled and that they are scoped to **All Users** per the baseline.

**Comment by tkol2022** [comment]
Here is an example of what a properly configured MS authenticator looks like in the Azure AD authentication methods page in the portal.

![image](https://github.com/cisagov/ScubaGear/assets/107076927/c732b8f5-a61d-4ee1-a587-9a9fa7421fde)

**Comment by tkol2022** [comment]
This should not affect the coding of this issue but I'm just documenting it here for historical references.

@nanda-katikaneni  and I tested the MS Authenticator settings for (show application name and geographic location). We found that, as long as MS Authenticator is Enabled and setup as per the 3.3 implementation requirements, the MS Authenticator app uses those features regardless of what value the tenant is set to in Authentication Methods > Manage Migration > migration status. In other words, the tenant does not have to be at the Migration Complete status for the authenticator configuration to work.
